### ISSUE TITLE
Swift 5.2: Additional false positive for `attribute` with `@autoclosure` parameter

### ISSUE THREAD

**Issue by zwaldowski** [realm/SwiftLint#3112]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

See previously #3097.

`attribute` hits for a function with attributes when it also accepts an `@autoclosure` parameter with Swift 5.2. It does not under Swift 5.1.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --config swiftlint.yml Suffer.playground/Contents.swift
Loading configuration from 'swiftlint.yml'
Linting Swift files at paths Suffer.playground/Contents.swift
Linting 'Contents.swift' (1/1)
/Users/zw/Downloads/Suffer.playground/Contents.swift:5:12: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports. (attributes)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.39.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Homebrew

* Paste your configuration file:

```yml
opt_in_rules:
  - attributes
attributes:
  always_on_same_line:
    ["@IBOutlet", "@IBAction", "@IBSegueAction", "@NSManaged"]
  always_on_line_above:
    ["@IBDesignable", "@IBInspectable", "@available", "@discardableResult"]
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

Nope.

* Which Xcode version are you using (check `xcodebuild -version`)?

Version 11.4 beta (11N111s)

* Do you have a sample that shows the issue?

```swift
enum Foo {

    // Violation.
    @inlinable
    public func show1(_ statement: @autoclosure() -> String) {
    }

    // No violation.
    @inlinable
    public func show2(_ statement: () -> String) {
    }

}
```
